gtest/src/manager.rs

```diff
@@ -30,17 +30,82 @@ use gear_core::{
 use std::collections::{BTreeMap, VecDeque};
 use std::time::{SystemTime, UNIX_EPOCH};
 
-#[derive(Clone, Debug)]
-pub(crate) enum ProgramState {
-    Initialized,
-    Uninitialized(Option<MessageId>),
-    FailedInitialization,
+#[derive(Debug)]
+pub(crate) enum Actor {
+    Initialized(Program),
+    // Contract: program is always `Some`, option is used to take ownership
+    Uninitialized(Option<MessageId>, Option<Program>),
```

Comment:
I bet you @breathx ask why `Option` for program here. You see, same program value is **moved** from `Uninitialized` to `Initialized while being under mutable reference. Will be happy to see neat alternatives.

Reply:
Where can I collect your bet? 😆 
This impl seems okay for me for current arch